source/agora/cli/multi/main.d

```diff
@@ -0,0 +1,305 @@
+            config.node.address = "127.0.0.1";
+
+            // Make sure that where the configuration file is located is the root directory of the node.
+            // The data directory on the node has been changed to an absolute path.
+            config.node.data_dir = buildNormalizedPath(dirName(file), config.node.data_dir);
+
+            // Multiple nodes simultaneously output logs to the console.
+            // To reduce complexity, only errors are printed.
+            // Other improvements are needed in the future.
+            config.logging.log_level = LogLevel.Error;
+
+            // Change the network address of the nodes.
+            string[] coverted_network;
+            foreach (address; config.network)
+                coverted_network ~= replaceAddress(address, "127.0.0.1");
+
+            Config coverted_config =
+            {
+                banman : config.banman,
+                node : config.node,
+                network : assumeUnique(coverted_network),
+                dns_seeds : config.dns_seeds,
+                quorum : config.quorum,
+                logging: config.logging,
+                admin: config.admin,
+            };
+
+            // Add
+            configs ~= coverted_config;
+        }
+        catch (Exception ex)
+        {
+            writefln("Failed to parse config file '%s'. Error: %s",
+                file, ex.message);
+            return 1;
+        }
+    }
+
+    // In the case of to connect to the node
+    if (args[1] == "test")
+    {
+        foreach (const ref config; configs)
+            testConection(config.getNodeAddress());
+
+        return 0;
+    }
+
+    // In the case of to start the node
+    NodeLauncher[] launchers;
+    foreach (const ref config; configs)
+    {
+        auto launcher = new NodeLauncher();
+        launcher.start(config);
+        launchers ~= launcher;
+    }
+
+    scope (exit)
+    {
+        foreach (elem; launchers)
+            elem.stop();
+    }
+
+    return runEventLoop();
+}
+
+
+/*******************************************************************************
+
+    Print out help for this process.
+
+*******************************************************************************/
+
+private void printDefaultHelp ()
+{
+    writeln("usage: agora-multi command [path] node0.yaml node1.yaml node2.yaml ...");
+    writeln("These are commands:");
+    writeln("   start : Starts the nodes");
+    writeln("   test : Test connections to the nodes");
+}
+
+
+/*******************************************************************************
+
+    Replace host name in network address
+
+    Params:
+      source = A network address
+      host = A new hostname
+
+    Returns:
+      A network address with changed hostname
+
+*******************************************************************************/
+
+private string replaceAddress(string source, string host)
```

Comment:
Wouldn't it be better to just parse the URL with the Vibe.d class and change the host ?

Reply:
https://vibed.org/api/vibe.inet.url/URL